Improved muon track validation (step1: base code deployment, no change to workflows)

cmsbuild opened the pull request:
This is the first step in deployment of an improved muon track validation code.
The strategy is agreed with the Muon POG and PPD conveners.

This PR contains new code and configuration files, introduced alongside with the current code and cfgs. The new code is not yet used in the standard workflows, i.e. nothing should change anywhere.

The second step will be the customization of a few configuration files (marked by a comment) to let the new and the default code to run (both) for a given 81X prerelease cycle in few selected muon RelVals. This will allow for a smooth transition between the current and the new validation code. The plan foresees to deploy the new validation at the prerelease following the one with duplicate validations. At that point the old code should be possibly removed.
As I will be on vacation for the next few weeks this is supposed to let others to implement the necessary customization to try to speed up things a bit.
Also this base code is convenient for other developments (new code for upgrades, which can be integrated afterwards).

As a side effect the main muon POG validation affects the HIN validation of muons. HIN should decide what to do during this intermediate period. (I am available to help out, I tried to update the HIN part as well).

The new code and configuration has been flagged by a prefix "New" in filenames, class/module names, when it is a duplicate of the current production code. 
Features:
- much clearer code, easier maintenance and developments, some technical improvements
- new plots for performance vs PU, charge misid ...
- removal of a few unused plots
- improved settings of the histogram binning/ranges for different types of muons. All such settings contained in a cfg file "histoParameters_cff.py".
- better choice of names for histograms, corrected labels etc...
- better use of selectors to clean up the cfgs
- slimmed/clearer cfgs as a consequence of the above changes
- kept legacy configurations for use of TrackAssociatorByPosition in separate files, avoiding to clutter the main cfgs as in the current version
- new default setting of the MuonAssociatorByHits to flag as fakes global muons with tracker and muon hits matched to different particles: this will produce sensible fake rates (much higher than what is obtained currently). Side note: the current configuration has always lived for historical reasons.

cheers
    Giovanni

Automatically ported from CMSSW_8_1_X #15153 (original by @abbiendi).

cmsbuild commented:
A new Pull Request was created by  @cmsbuild  for CMSSW_9_0_X.

It involves the following packages:

Configuration/StandardSequences
HLTriggerOffline/Muon
SimMuon/MCTruth
Validation/Configuration
Validation/RecoHI
Validation/RecoMuon


@civanch, @cvuosalo, @mdhildreth, @dmitrijus, @cmsbuild, @franzoni, @slava77, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @battibass, @makortel, @kkrajczar, @abbiendi, @GiacomoSguazzoni, @jhgoh, @VinInn, @Martin-Grunewald, @calderona, @HuguesBrun, @rylanC24, @rovere, @trocino, @dgulhan, @rociovilar this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented:
this feature was pending in 81X for 2 months since L2 signatures were collected.
Is it still needed in the present implementation?

cmsbuild commented:
+1
Tested at: 1ce47da36a5f7ff590d12e0cf976c5b7ac9f5094
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16646/16434/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16646/16434/summary.html

Alternative comparison was/were failed for workflow(s):
50202.0

civanch commented:
+1

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/16646 1ce47da
... with the same considerations as for https://github.com/cms-sw/cmssw/pull/15153 1ce47da

dmitrijus commented:
+1

rovere commented on Validation/RecoMuon/plugins/NewMuonTrackValidatorBase.h:
```diff
@@ -0,0 +1,285 @@
+    ignoremissingtkcollection_(pset.getUntrackedParameter<bool>("ignoremissingtrackcollection",false))
+
+    {
+      dbe_ = edm::Service<DQMStore>().operator->();
```
Why do you need to hold a pointer to DQMStore? In the multithread world this is asking for trouble. I saw no real usage (correct) directly of it: I'd suggest to remove it, so the next-to-come developer is not tempted to use it...

rovere commented on Validation/RecoMuon/plugins/NewMuonTrackValidatorBase.h:
```diff
@@ -0,0 +1,285 @@
+
+ protected:
+
+  DQMStore* dbe_;
```
Same comment here.

dmitrijus commented:
It looks like it is never used - I agree with Marco, should be deleted.

ferencek committed:
switched to v4 training for ak8 jets

ianna committed:
Test if GEM record is present before building GEM geometry

ianna committed:
Add GEM geometry producer to configuration to retrieve GEM record from DB

ianna committed:
Add GEM reco geometry producer

ianna committed:
Add GEM payload producers to 2017 payload scripts

ianna committed:
Add GEM payload producers to 2017 payload scripts

pietverwilligen committed:
revert back ME0DetId numbering: if NEtaPart=1 => rollId = 0

rekovic committed:
Use L1T Stage2 products for DIGI2RAW & RAW2DIGI in Phase2 era.

knoepfel committed:
Add fillDescriptions static member functions.

bapavlov committed:
iRPC 2D readout

bapavlov committed:
iRPC 2D readout

bapavlov committed:
iRPC 2D readout

bapavlov committed:
some comments removed

cmsbuild committed:
Merge pull request #17232 from davidlt/clean-noexcept-caloonline

Replace throw dynamic exception specifiers with noexcept

cmsbuild committed:
Merge pull request #17207 from ggovi/condcore-utilities-o2o-5-90X

Improvements for o2o tools

cmsbuild committed:
Merge pull request #17147 from davidlt/gcc630-cxx11abi-fix-oracledb

Avoid Oracle OCCI APIs which work with std::string in C++11 ABI mode

cmsbuild committed:
Merge pull request #17213 from davidlt/clean-onlinedb-noexcept2

Replace throw specifier (deprecated) with noexcept (C++11)

davidlange6 committed:
Merge pull request #17226 from matz-e/hcal-tp-robustness

HCAL TP: Address outstanding comment from #17171

Commit:
Replace buf/sprintf to std::to_string

According to compiler (GCC 7) we need at least 11 bytes in the buffer.
Replace buf + sprintf (unsafe) with std::to_string.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

rekovic committed:
Add back simHcalTrigPrimitiveDigis and simHcalTTPDigis into hcalDigiSequence

cmsbuild committed:
Merge pull request #17205 from bendavid/multifitped_90x

Extended Multifit functionality

walterjr committed:
2015LowPUMixProfile90X

mtosi committed:
trackHandle --> trackCollection

VinInn committed:
use refitted in strip as well

VinInn committed:
do not use refitted for tracking

pietverwilligen committed:
undo previous change, keep roll numbering starting at 1

pietverwilligen committed:
do not use roll() equal or different from 0 to distinguish M1 and M2, but use nPartitions > 1 instead

VinInn committed:
use refitted in strip as well

Commit:
2017 scenario: HCAL digiitzation tags updated to roll HE back to phase0 ( https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/2801.html )

VinInn committed:
make iteration to iterate

matz-e committed:
Increase robustness of HCAL TP LUT generation.

The current code crashes for Phase2.  Instead of generating all possible
DetIds and testing which ones are allowed, it is a lot cleaner to just
iterate over the produced metadata ids.

VinInn committed:
derefence handle not nulptr

cmsbuild committed:
Merge pull request #17255 from davidlt/fix-buf-to_string

Replace buf/sprintf to std::to_string

cmsbuild committed:
Merge pull request #17243 from davidlt/fix-clang-warn

Clean Clang warnings regarding hiding overloaded virtual functions

kpedro88 committed:
remove now-unnecessary customization for 2017

emanueledimarco committed:
Merged refs/pull/17205/head from repository cms-sw

depasse committed:
Reading TPG parameters from TPG_beamv6_trans_xxx_spikekill.txt file

cmsbuild committed:
Merge pull request #17099 from bsunanda/Run2-hcx97

bsunanda:Run2-hcx97 Bug fixes for Hcal geometry routines

abbiendi committed:
Merged devNewValidation_81X from repository abbiendi

kpedro88 committed:
fixes for DB testing code

abbiendi committed:
remove pointer to DQMStore

cmsbuild committed:
Merge pull request #17246 from intrepid42/weights_90X

Read compressed LHE weights

cmsbuild committed:
Merge pull request #17221 from mtosi/fixTrackCollection

edm::View<reco::Track> instead of reco::TrackCollection in DQM/TrackingMonitor

bapavlov committed:
Update muonRPCDigis_cfi.py

bapavlov committed:
Update RPCSimModelTiming.cc

bapavlov committed:
Update RPCSimModelTiming.cc

bapavlov committed:
Update RPCSimModelTiming.cc

bapavlov committed:
Update RPCSimModelTiming.cc

davidlange6 committed:
Revert "CTPPS: miniAOD"

davidlange6 committed:
Merge pull request #17265 from cms-sw/revert-17162-ctpps_miniaod

Revert "CTPPS: miniAOD"

bapavlov committed:
Merged iRPC_2D from repository bapavlov

bapavlov committed:
std::cout changed with edm::LogInfo

matz-e committed:
Filter out invalid ids.

Commit:
Replace TString with std::string to avoid C++17 issues

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Enable possibility of running GEN-SIM for some 2017 geometry and choosing Plan0/1/36 at DIGI-RECO level

Commit:
Interface change in view of handling special RBX

Commit:
Remove char buffer and sprintf with std::string/std::to_string

Originally I expected ROOT classes to support std::string, but they
actually take const char * instead thus I had to use .c_str(), which is
not the nicest code. Hopefully ROOT classess makes a copy and not just
saves the pointer internally as std::string will be temporaries.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Replace bool with unsigned int

This was picked by GCC 7:

    error: use of an operand of type 'bool' in 'operator++' is forbidden
    in C++1z

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Resolve -Werror=format-truncation= issues

By default %f precision is 6 digits after comma. Same applies to
std::to_string() and operator<< on std::basic_ostream.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Replace throw specifier with noexcept

Throw was deprecated in C++11 and removed in C++17.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Replace boost stuff with std where possible

We should avoid boost where possible and rely on C++ standard library.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Replace forward decl for PFTrackTransformer with full header incl

Looks like now `std::unique_ptr` uses `sizeof(_Tp)` thus we need to know
full class layout.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Replace deprecated throw with noexcept

throw dynamic exception specifier was deprecated in C++11 and removed in
C++17.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

ggovi committed:
A few fixes for error handling in connection strings

ggovi committed:
Added o2oTest option for the copy tag function

rovere committed:
Update RecoMaterial

PR 17245 changed the material budget for the pixel detector.
This PR derives the reconstruction material from the updated
simulation. No significant effects have been seen in the
final tracking performances. Only pixel-related material
budget has been retuned in this PR, on purpose.

emanueledimarco committed:
optional use of max-sample algo for gain-switch hits

jfernan2 committed:
Removed producers from source configuration cffs

rekovic committed:
Missed: Legacy Raw2Digi also if not phase2_common.

rekovic committed:
In case of phase2_common era, do simGtExtFakeStage2Digis, like in case of stage2L1Trigger.

rekovic committed:
Merged refs/pull/17260/head from repository cms-sw

makortel committed:
Use also TrackingParticle::numberOfHits()==0 in QuickTrackAssociatorByHits

In TrackingTruthAccumulator, the numberOfHits is calculated from a
subset of the SimHits of the SimTracks of a TrackingParticle
(essentially limiting the processType and particleType to those of the
"first" hit, and particleType to the pdgId of the SimTrack). But, here
the association between tracks and TrackingParticles is done with
*all* the hits of TrackingParticle, so we should not rely on the
numberOfHits() calculated with a subset of SimHits.

Commit:
2017 scenario: HCAL digiitzation tags RE-updated to roll HE back to phase0 ( https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/2802/1.html ) ; 2018 scenario: change all HCAL tags with a better naming and the identical information as prior

rekovic committed:
Add forgotten import of phase2_common

makortel committed:
Drop TrackingParticle::numberOfHits()==0 check also from the ntuple

makortel committed:
Sort SimHits by their time of flight for the number of hits calculation

Because the processType and particleType are taken from the "first"
SimHit of a SimTrack, it is important to get the "first" right.
Previously it was the one with the smallest index in the SimHit
vector, but this fails for Phase2 tracker where also the outer tracker
(OT) SimHits are in the same event products as pixel SimHits, and for
some reason the OT SimHits are inserted before the pixel SimHits.

I believe using the time of flight is more robust way to decide which
SimHit is "first".

makortel committed:
Small performance optimization

civanch committed:
define default dark_factor

civanch committed:
extended printout

cmsbuild committed:
Merge pull request #17273 from davidlt/boost-noexcept-xhist

Cleanup throw specifier and boost stuff

davidlange6 committed:
Merge pull request #17271 from davidlt/bool-un-int

Replace bool with unsigned int

cmsbuild committed:
Merge pull request #17270 from davidlt/clean-sprintf-val

Remove char buffer and sprintf with std::string/std::to_string

cmsbuild committed:
Merge pull request #17237 from davidlt/framework-noexcept-false

Add dtors with noexcept(false) to match C++ requirements

cmsbuild committed:
Merge pull request #17269 from bsunanda/Run2-hcx99

bsunanda:Run2-hcx99 Interface change in view of handling special RBX

cmsbuild committed:
Merge pull request #16598 from nancymarinelli/TP_PhaseII_V0

Tp phase ii v0

davidlange6 committed:
Merge pull request #17281 from walterjr/2015LowPUMixProfile90X

2015LowPUMixProfile90X

davidlange6 committed:
Merge pull request #17272 from davidlt/fix-format-trunc

Resolve -Werror=format-truncation= issues

cmsbuild committed:
Merge pull request #17252 from bapavlov/iRPC_2D

iRPC_2D

cmsbuild committed:
Merge pull request #17210 from schneiml/matbudget-fix-ca-pre3

Material Budget Tool Fixes

davidlange6 committed:
Merge pull request #17267 from davidlt/replace-TString-string-val

Replace TString with std::string to avoid C++17 issues

cmsbuild committed:
Merge pull request #17235 from fwyzard/HLT_customisations_for_HCAL_Phase_I

HLT customisations for the Phase I HCAL upgrades

fwyzard committed:
move migration to HBHEPhase1Reconstructor to the common run2_HCAL_2017 era

Commit:
Add possibility to switch on/off neutral density filter correction at SIM and DIGI levels

cmsbuild committed:
Merge pull request #17284 from fwyzard/HLT_customisations_for_HCAL_Phase_I

move migration to HBHEPhase1Reconstructor to the common run2_HCAL_2017 era

mdhildreth committed:
add base configuration to allow arbitrary pileup functions on a stable config

jhgoh committed:
Merged refs/pull/17252/head from repository cms-sw

jhgoh committed:
RPCRecHit x-position correction for endcap

Commit:
bugfix for a feature not normally used

ianna committed:
Add payload scripts for 2018 geometry scenario

ianna committed:
Update 2018 geometry scenario to pick up newest changes added to 2017

ianna committed:
Remove obsolete comment

ianna committed:
Redice rail width and move it by a mm

rovere committed:
Silent warning due to missing templates in PhaseII

The materialDumpAnalyser uses internally a refitter to
extract the angle of incidence of the track with respect to
the sensor surface. The refitter uses 'angleAndTemplate',
that in PhaseII is missing. This PR drops the template
part.

abbiendi committed:
merge with recent independent developments

Commit:
Merged devNewValidation_90X from repository abbiendi

kpedro88 committed:
fix regression in SimCalorimetry_EventContent

kpedro88 committed:
fix era import

kpedro88 committed:
only run new EB trig prims in phase2

kpedro88 committed:
keep ecal unsuppressed digis for phase2

kpedro88 committed:
fix entry in sequence has no label

capalmer85 committed:
add selecting bunch feature to estimate PU script - 90X

cmsbuild committed:
Merge pull request #17283 from civanch/fix_for_exotica_sim

Fix for exotica simulation

kpedro88 committed:
remove unused vars

davidlange6 committed:
Merge pull request #17170 from makortel/reworkSeeding_pixelTracks_lastpiece

Migrate PixelTrackProducer and HLT to new seeding framework

mdhildreth committed:
Merged refs/pull/17281/head from repository cms-sw

mdhildreth committed:
Modify new PU specification to import/conform to new base configuration

manoni committed:
Implementation of several plots in SiStripMonitorCluster package: number of cluster trend in wheel/layer, number of cluster per ring and per layer. Change in apv charge median plot switch in SiStripMonitorDigi package

cmsbuild committed:
Merge pull request #17287 from mdhildreth/pileupBaseFunction9

add base configuration to allow arbitrary pileup functions on a stabl…

davidlange6 committed:
Merge pull request #17253 from aperloff/fixJetMETCorrectionsConfigurationCorrectedJetProducers

Thread Safe Jet Correctors

Commit:
First set of changes to make the customization run in pp

cmsbuild committed:
Merge pull request #17268 from bsunanda/Run2-hcx98

bsunanda:Run2-hcx98 Enable possibility of running GEN-SIM for some 2017 geometry and choosing Plan0/1/36 at DIGI-RECO level

rekovic committed:
Add missing Stage2 sim products to L1TriggerFEVTDEBUG and shorten keep *Stage2Digis.

cmsbuild committed:
Merge pull request #17299 from capalmer85/add-selbx-estPU-90X

add selecting bunch feature to estimate PU script - 90X

cmsbuild committed:
Merge pull request #17215 from knoepfel/convert-JetConstituentSelector

Convert JetConstituentSelector template from legacy to stream filter

cmsbuild committed:
Merge pull request #17015 from gartung/DQM-HcalCommon-Tasks-fix-clang-warnings

DQM-HcalCommon+HcalTasks: fix clang warnings hides overloaded virtual function

cmsbuild committed:
Merge pull request #17097 from suchandradutta/Phase2Digitizer_90X_061222

Update of the Phase2 Tracker Digitizer Validation suite

cmsbuild committed:
Merge pull request #17275 from davidlt/clean-throw-PFClusterTools

Replace deprecated throw with noexcept

cmsbuild committed:
Merge pull request #17274 from davidlt/PFTrackTransformer

Replace forward decl for PFTrackTransformer with full header incl

Commit:
Add a few test to verify Plan-1 geometry

davidlange6 committed:
Merge pull request #17198 from kpedro88/Phase1-HE14

Phase0 HE in 2017, Phase1 HE in 2018

cmsbuild committed:
Merge pull request #17208 from schneiml/phase1pixeldqm-post-christmas-version

Phase1 Pixel DQM PixelMaps, new TrackEfficiencies, removed Harvesting, new Plots and various Bugfixes

davidlange6 committed:
Merge pull request #17297 from kpedro88/Phase2-WF32

Fix config for Phase2 EB trig prim digis

davidlange6 committed:
Merge pull request #17310 from bsunanda/Run2-hcx100

bsunanda:Run2-hcx100 Add a few test to verify Plan-1 geometry

slava77devel committed:
put unscheduled customs in a proper order: external methods are to be applied after all internal (miniAOD) are done

Conflicts:
	Configuration/Applications/python/ConfigBuilder.py

boudoul committed:
Adding ng Stubs and Track Trigger Sequence

jhgoh committed:
seems to have flipped sign in x correction

bbockelm committed:
Fix string quoting for CLI.

Previously, the quoting of the CPU model output did not obey
ClassAd rules for a string.  This implements a reasonable subset
(throwing out the difficult parts; if a CPU model contains a
double-quote character, we'll lose it).

davidlange6 committed:
Merge pull request #17312 from slava77/CMSSW_9_0_X_2017-01-27-1100/sign835/cBuilderUncheduledCustoms

put unscheduled customs in order: external methods done applied after all internal (miniAOD) are done

VinInn committed:
merge with HEAD

VinInn committed:
Merged TrajInTrack9X from repository VinInn

VinInn committed:
Merge branch 'CMSSW_9_0_X' into from-CMSSW_9_0_X_2017-01-29-1100

VinInn committed:
fix traj position

Commit:
Added HLT to customization

Commit:
Minor fix (re-enabling selectors import)

ggovi committed:
Various fixes for error handling

Commit:
geometry updates: Tracker and Gem affecting 2017 and 2018 scenarios

cmsbuild committed:
Merge pull request #17294 from cms-btv-pog/BoostedDoubleSVTaggerV4-PR_from-CMSSW_9_0_0_pre2

Switching BoostedDoubleSV tagger to v4 training for AK8 jets

cmsbuild committed:
Merge pull request #17279 from emanueledimarco/multifit_gainswitch_maxsample_90X

Optional use of Max-Sample algorithm for gain-switch cases

Commit:
Add customise on RAW-RECO

cmsbuild committed:
Merge pull request #17115 from ianna/gem-slice-in-2017-scenario

GEM Slice Test in 2017 Scenario

cmsbuild committed:
Merge pull request #17156 from ianna/fpix-overlap-fix-v2

Remove Overlaps in FPix

cmsbuild committed:
Merge pull request #17199 from ianna/tob-connect56-overlap-fix

TOB Connectors Overlap Bugfix

cmsbuild committed:
Merge pull request #17280 from rovere/recoMaterialTuning

Update RecoMaterial

cmsbuild committed:
Merge pull request #17245 from schneiml/pixel-matbudget-changes-st

Pixel Material Budget Changes (updated)

cmsbuild committed:
Merge pull request #17250 from pietverwilligen/90X_NewME0GeometryBuilder_bugfix

ME0ReDigitizer not using roll==0 to distinguish between M1 and M2 geometry

syuvivida committed:
replace 1e-6 with zero

davidlange6 committed:
Merge pull request #17261 from ianna/tib-rail-bottom-overlap-fix

TIB Rail Bottom Overlap Bugfix

ianna committed:
Move to configuration by Era. Include GEM for run2 in 2017.

Commit:
Including cosmics and FastSim

cmsbuild committed:
Merge pull request #17301 from yiiyama/ecaldigifix-90

Bugfix EcalSimProducer for APD simulation

davidlange6 committed:
Merge pull request #17314 from boudoul/trackTriggWF_90X

Adding Stubs within Track Trigger Sequence (Phase2 Tracker)

VinInn committed:
add return process

Commit:
Cleanup of unnecessary cms sequence placeholders

Dr15Jones committed:
Allow restarting MessageSender

Need to be able to reset a static data structure in the case where
multiple CmsRun python instances are created one after the other.

Dr15Jones committed:
Properly initialize PythonEventProcessor

The edm::EventProcessor held by PythonEventProcessor need some
standard Services defined in order function properly. By moving
the list of Services to its own function, CmsRun and cmsRun.cc
can share the same list.

Dr15Jones committed:
Fix pickling of SortedKeysDict

When trying out mpi4py's serialization of cms.Process using pickling
trying to read-back a SortedKeysDict failed. The unpickling process
is not guaranteed to call __init__ which meant self.list was not
being initialized.

davidlange6 committed:
turn off 2017 gen sim recycle for now

davidlange6 committed:
Merge pull request #17332 from davidlange6/remove2017recylc

turn off 2017 gen sim recycle for now

cmsbuild committed:
Merge pull request #17309 from rekovic/PR90x_L1T-FEVTDEBUG-EventContent-fix

90x L1T FEVTDEBUG fix missing Stage2 products

cmsbuild committed:
Merge pull request #17316 from bbockelm/fix_classad_quoting

Fix string quoting for condor_chirp CLI.

cmsbuild committed:
Merge pull request #17328 from Dr15Jones/fixPythonCmsRun

Add missing initializations needed by python CmsRun

cmsbuild committed:
Merge pull request #17329 from Dr15Jones/pickledSortedDict

Fix pickling of SortedKeysDict

davidlange6 committed:
Merge pull request #17319 from franzoni/gf_moreGeoUpdates_2017-2018_from-CMSSW_9_0_X_2017-01-30-1100

geometry updates: Tracker and Gem affecting 2017 and 2018 scenarios

davidlange6 committed:
Merge pull request #17324 from ianna/gem-reco-geometry-from-DB-in-2017-era

GEM Reco Geometry from DB in 2017 Era

VinInn committed:
use View in refitter

VinInn committed:
improve example and traj analyzer

kpedro88 committed:
handle testnumbering in GlobalHits validation

kpedro88 committed:
fix testnumbering handling in HcalDigisValidation

lorusso7 committed:
Update to Madgraph 2.4.2

lorusso7 committed:
Update to Madgraph 2.4.2

lorusso7 committed:
Update to Madgraph 2.4.2

lorusso7 committed:
Update to Madgraph 2.4.2

Commit:
alignment (asy and real) Consistent with Fpix updated Thickness (290 instead of 300 mum)

cmsbuild committed:
Merge pull request #17320 from syuvivida/90X_genfilterinfo_fix

Fix of GenFilterInfo in 90X release

knoepfel committed:
White space changes only.

knoepfel committed:
Remove unnecessarily specified d'tor.

knoepfel committed:
Convert LogErrorHarvester to global::EDProducer.

cmsbuild committed:
Merge pull request #17278 from ggovi/condcore-utilities-tools-improvements-3

conddb tools improvements

cmsbuild committed:
Merge pull request #17248 from cms-l1t-offline/pr90x-L1TStage2-forPhase2

Pr90x L1T Stage2 for Phase2

davidlange6 committed:
more workflows that can not use recycled gensim at this point

davidlange6 committed:
Merge pull request #17337 from franzoni/gf_alignmentConsistentWithFpixThickness-from-CMSSW_9_0_X_2017-01-31-1100

alignment (asy and real) Consistent with Fpix updated Thickness (290 instead of 300 mum)

gartung committed:
Changes needed to compile with macOS clang and libc++

davidlange6 committed:
Merge pull request #17341 from davidlange6/more20178recycleremovals

more workflows that can not use recycled gensim at this point

gartung committed:
Changes needed to compile with macOS clang and libc++

gartung committed:
Changes needed to compile with macOS clang and libc++

gartung committed:
Changes needed to compile with macOS clang and libc++

davidlange6 committed:
Merge pull request #17098 from VinInn/TrajInTrack9X

Remove Trajectory from event

davidlange6 committed:
Merge pull request #17335 from kpedro88/Phase1-HE15

Fix or suppress warnings related to Phase0 HE in 2017

VinInn committed:
migrate HI as well

davidlange6 committed:
remove comparisons of nonnull objects against 0

davidlange6 committed:
Merge pull request #17363 from VinInn/Fix17098

Fix 17098

ggovi committed:
Removed CondCore/ORA package

VinInn committed:
make TrackExtra fully persistent because of FastSimPU

davidlange6 committed:
Merge pull request #17367 from VinInn/TrackExtraPersistent

make TrackExtra fully persistent because of FastSimPU

Dr15Jones committed:
The new gcc ABI for C++11 uses different namespace

The gcc ABI for C++11 internally uses the namespace std::__cxx11
for the standard library containers. The friendly class name
generator now knows about that naming in order to keep the
branch names the same as before the ABI change.

cmsbuild committed:
Merge pull request #17116 from calabria/NewME0Validation90X

New ME0 validation

cmsbuild committed:
Merge pull request #17249 from depasse/TPGfromFile

Reading TPG parameters from file

cmsbuild committed:
Merge pull request #17304 from manoni/CMSSW_9_0_ROOT6_X_2017-01-16-2300_clusTrend9bis

Modification/Addition in SiStripMonitorDigi and SiStripMonitorCluster packages

cmsbuild committed:
Merge pull request #17336 from lorusso7/Madgraph_2.4.2_cards

Madgraph 2.4.2 cards

cmsbuild committed:
Merge pull request #17338 from makortel/quickTrackAssocRemoveNumberOfHits

Remove check on TrackingParticle::numberOfHits()==0 from QuickTrackAssociatorByHits

cmsbuild committed:
Merge pull request #17354 from gartung/Fireworks-Core-macos-clang-fixes

Fireworks/Core : Changes needed to compile with macOS clang and libc++

cmsbuild committed:
Merge pull request #17290 from rovere/SilentTemplateWarnInMatDumper

Silent warning due to missing templates in PhaseII

cmsbuild committed:
Merge pull request #17355 from gartung/Geometry-CaloGeometry-macos-clang-fixes

Geometry/CaloGeometry : Changes needed to compile with macOS clang and libc++

davidlange6 committed:
Merge pull request #17366 from ggovi/condcore-remove-ora

Remove CondCore/ORA package

davidlange6 committed:
Merge pull request #17131 from ianna/gem-in-2017-payload-scripts

GEM Reco Geometry Payload Writer Added to 2017 Scripts

davidlange6 committed:
Merge pull request #17291 from ianna/extended-2018-geometry-scenario

Extended 2018 Geometry Scenario Update

cmsbuild committed:
Merge pull request #16739 from jfernan2/dropDDU_fromDTDQMtrigger_90X

Changes to disable DDU trigger plots in DT DQM Online Trigger checks …

davidlange6 committed:
Merge pull request #17359 from gartung/SimDataFormats-RandomEngineState-macos-clang-fixes

SimDataFormats/RandomEngineState : Changes needed to compile with macOS clang and libc++

Commit:
Update the gap value in GE21 design for TDR_Dev version

cmsbuild committed:
Merge pull request #16638 from fthyssen/rpc-amc-unpacker-conditions

RPC TwinMux and OMTF/CPPF unpacker conditions

Commit:
HI sequences updated

gartung committed:
changes needed to compile on macOS sierra with system clang and libc++

gartung committed:
wrap change for rootcling

makortel committed:
Ignore upfront SimHits with particleType different from SimTrack pdgId

These are anyway ignored from the number of hit calculation, so let's
ignore them from the "first hit" deduction as well.

cmsbuild committed:
Merge pull request #17374 from bsunanda/Phase2-gem22

bsunanda:Phase2-gem22 Update the gap value in GE21 design for TDR_Dev version

cmsbuild committed:
Merge pull request #17377 from knoepfel/convert-LogErrorHarvester

Convert edm::LogErrorHarvester from legacy to global EDProducer.

cmsbuild committed:
Merge pull request #17342 from gartung/FWCore-Framework-macos-clang-fixes

FWCore/Framework : Changes needed to compile with macOS clang and libc++

cmsbuild committed:
Merge pull request #17371 from Dr15Jones/gccAbiChange

The new gcc ABI for C++11 uses different namespace

gartung committed:
add const

boudoul committed:
Removing cout from code used in prod

kpedro88 committed:
temp fix 2017 WF nums by skipping ranges

davidlange6 committed:
Revert "Pr90x L1T Stage2 for Phase2"

davidlange6 committed:
Merge pull request #17392 from cms-sw/revert-17248-pr90x-L1TStage2-forPhase2

Revert "Pr90x L1T Stage2 for Phase2"

davidlange6 committed:
Merge pull request #17391 from kpedro88/Phase1-WF2

temp fix 2017 WF nums by skipping ranges

davidlange6 committed:
Merge pull request #17390 from boudoul/silent

Removing 'cout' comments  from code used in prod (phase2 Track Trigger)

cmsbuild committed:
Merge pull request #17382 from makortel/fixTrackingParticleNumberOfHits

Fix calculation of TrackingParticle::numberOfHits()

cmsbuild committed:
Merge pull request #17380 from gartung/DataFormats-CaloTowers-macos-clang-fixes

DataFormats/CaloTowers : changes needed to compile on macOS sierra with system clang and libc++

cmsbuild committed:
Merge pull request #17364 from davidlange6/fix630warnings

remove comparisons of nonnull objects against 0

cmsbuild committed:
Merge pull request #17334 from VinInn/RefitWithView

use edm::View in TrackRefitter

cmsbuild committed:
Merge pull request #17289 from jhgoh/RPCRecHitXCorrection

RPCRecHit local x position correction

Commit:
Split associators and NewAssociators

abbiendi committed:
Merged devNewValidation_90X_Customise from repository battibass

slava77 commented:
@dmitrijus 
@battibass reminded me recently that this code is needed in the release.
Please see that it makes it or follow up with @abbiendi and @battibass to see alternative solutions

dmitrijus commented:
@slava77 
A single unused pointer is not an issue, it was copied from older code anyway. If this is urgent, please merge, I already signed.
However, there was no reply from author, which is "worrying".

abbiendi commented:
Hi, I have already fixed it (thanks for the comment). I was waiting as now we have almost ready a major update. The customization which was foreseen as second step for the real switch on of the code is ready, thanks to Carlo Battilana. In addition I have synchronized the "New" version of the validation to the parallel improvements that have been going on (by Cesare Calabria and others) since the time of my initial PR. Given the large number of changes I am not sure what is the best way to proceed: to open a new PR which should absorb the current one or to update the current PR. The first solution would be easier for me, just I would like not to lose the initial description of the PR. Please tell me what is better to do in your opinion. thanks,  Giovanni

On Feb 7, 2017, at 5:08 PM, Dmitrijus wrote:


@slava77<https://github.com/slava77>
A single unused pointer is not an issue, it was copied from older code anyway. If this is urgent, please merge, I already signed.
However, there was no reply from author, which is "worrying".

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/16646#issuecomment-278047039>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFZNr2ORza3-9acSpk5awFA58AhxU8amks5raJcXgaJpZM4Kzn_n>.

abbiendi committed:
customization done: remove old comments

abbiendi committed:
revert back the old configuration files (to stay untouched)

abbiendi committed:
split completely the new validation sequences from the old ones

abbiendi committed:
remove unused configuration files

abbiendi committed:
fix printouts in the customization functions

abbiendi committed:
fix comments

abbiendi committed:
move mixing module to the prevalidation sequence

abbiendi committed:
cleanup of the customization function

abbiendi commented:
I'm updating the PR with all the wanted features:
- removed the pointer to DQMStore (as requested)
- synchronized the New validation code to the current code for parallel developments in the last few months (for upgrades)
- added a customization function to activate the New validation for the temporary period of coexistence with the old one (thanks to Carlo Battilana)

cmsbuild commented:
Pull request #16646 was updated. @ghellwig, @cvuosalo, @arunhep, @cerminar, @monttj, @cmsbuild, @rekovic, @franzoni, @slava77, @ggovi, @mmusich, @mulhearn, @davidlange6 can you please check and sign again.

slava77 commented:
-1 

needs a rebase
https://twiki.cern.ch/twiki/bin/viewauth/CMS/SWGuideHATS2016Git#CMSSW_rebasing_recipe

abbiendi commented:
couldn't I open a new PR in 90X ?

abbiendi commented:
and copy the initial description and the last one about this update ?

slava77 commented:
On 2/10/17 9:09 AM, abbiendi wrote:
> and copy the initial description and the last one about this update ?

you can open another PR
or make a new branch with the same name based on 90X and push --force to
the remote to update this PR.
or make a rebase.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16646#issuecomment-279002452>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbl7z0wu9dQiFTTsPbs7qKXpaDHGhks5rbJm7gaJpZM4Kzn_n>.
>

abbiendi commented:
thanks Slava ! 
I prefer the first option because I tested everything in a clean workdir, based on 900pre4, so it should be straightforward.

Also I'd like to understand why I ended up with this messy PR, with thousands of changed files.
Before my last "push", the PR16646 showed just 28 changed files w.r.t. 900pre4, as I can see from a local area where I downloaded the PR:

[giovanni@lxplus0051 src]$ pwd
/afs/cern.ch/work/g/giovanni/devNewValidation_90X/PR16646/CMSSW_9_0_0_pre4/src
[giovanni@lxplus0051 src]$ git diff -r CMSSW_9_0_0_pre4 --name-only
Configuration/StandardSequences/python/ValidationCosmics_cff.py
Configuration/StandardSequences/python/Validation_cff.py
HLTriggerOffline/Muon/python/HLTMuonPostVal_cff.py
HLTriggerOffline/Muon/python/HLTMuonVal_cff.py
SimMuon/MCTruth/python/NewMuonAssociatorByHits_cfi.py
Validation/Configuration/python/globalValidationCosmics_cff.py
Validation/Configuration/python/globalValidation_cff.py
Validation/Configuration/python/postValidation_cff.py
Validation/RecoHI/python/HarvestingHI_cff.py
Validation/RecoHI/python/NewMuonValidationHeavyIons_cff.py
Validation/RecoHI/python/globalValidationHeavyIons_cff.py
Validation/RecoMuon/plugins/NewMuonTrackValidator.cc
Validation/RecoMuon/plugins/NewMuonTrackValidator.h
Validation/RecoMuon/plugins/NewMuonTrackValidatorBase.h
Validation/RecoMuon/plugins/SealModule.cc
Validation/RecoMuon/python/NewAssociators_cff.py
Validation/RecoMuon/python/NewMuonTrackValidator_cff.py
Validation/RecoMuon/python/NewMuonTrackValidator_cfi.py
Validation/RecoMuon/python/NewMuonValidationHLT_cff.py
Validation/RecoMuon/python/NewMuonValidation_cff.py
Validation/RecoMuon/python/NewPostProcessorHLT_cff.py
Validation/RecoMuon/python/NewPostProcessor_cff.py
Validation/RecoMuon/python/NewSelectors_cff.py
Validation/RecoMuon/python/histoParameters_cff.py
Validation/RecoMuon/python/tabp_associators_cff.py
Validation/RecoMuon/python/tabp_muonValidationHLT_cff.py
Validation/RecoMuon/python/tabp_muonValidation_cff.py
Validation/RecoMuon/test/macro/NewTrackValHistoPublisher.C

The PR was created automatically in 90X, porting my previous PR from 81X. In fact the branch name had remained the same:  "devNewValidation_81X", and I did not like it at least because "81X" was misleading. 
For my "push" I followed the hint from this github page which says:
"Add more commits by pushing to the devNewValidation_81X branch on abbiendi/cmssw."
Now to understand what happened I imagine that the branch where I pushed was still the old one, based on 81X, while I believed to be automatically based on 90X.
(otherwise also the automatic porting from 81X to 90X should have met the same trouble, isn't it ?).  

sorry for the long chat...

slava77 commented:
On 2/10/17 3:00 PM, abbiendi wrote:
> thanks Slava !
> I prefer the first option because I tested everything in a clean
> workdir, based on 900pre4, so it should be straightforward.
> 
> Also I'd like to understand why I ended up with this messy PR, with
> thousands of changed files.
> Before my last "push", the PR16646 showed just 28 changed files w.r.t.
> 900pre4, as I can see from a local area where I downloaded the PR:
> 

I can guess that your 900pre4 branch was a result of "cms-merge-topic
16646",
which leads to the symptoms we see here.
git web doesn't match the merged part of going from the old base to
900pre4 with something actually in the release and displays the
difference with respect to the old base commit.
The unrelated differences displayed are in this sense fake.
The way to avoid this is not to use "cms-merge-topic xyz" to advance
development from an old release to the new one.

abbiendi commented:
yes, I used cms-merge-topic to work localy, first to get my PR on top of CMSSW_9_0_0_pre4 (with "git cms-merge-topic abbiendi:devNewValidation_81X"), then also to get changes from Carlo which have been merged. At the end I pushed the local branch with all the changes after all my tests to the branch associated to the PR.
At least you understand what was going on, I admit that still I don't understand. Anyway what was then the best way of doing ? Submitting a new PR in 90X ? (indeed I was asking for advice as I feared there would be troubles).

abbiendi commented:
I just opened a new PR replacing this one (which can be closed)

slava77 commented:
On 2/10/17 3:40 PM, abbiendi wrote:
> yes, I used cms-merge-topic to work localy, first to get my PR on top of
> CMSSW_9_0_0_pre4 (with "git cms-merge-topic
> abbiendi:devNewValidation_81X"), then also to get changes from Carlo
> which have been merged. At the end I pushed the local branch with all
> the changes after all my tests to the branch associated to the PR.


> At least you understand what was going on, I admit that still I don't
> understand. Anyway what was then the best way of doing ? Submitting a
> new PR in 90X ? (indeed I was asking for advice as I feared there would
> be troubles).

For better back-tracking to keep track of the past review thread
 it would be better to keep the updates in the old PR.

It's fine with a new PR.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16646#issuecomment-279096276>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbu6s8IQYpvLIGqFgWRl2CDMhppEvks5rbPVVgaJpZM4Kzn_n>.
>
